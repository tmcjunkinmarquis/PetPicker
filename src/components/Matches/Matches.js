import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './Matches.css';
import IndividualMatch from '../IndividualMatch/IndividualMatch';
import { fetchMatches } from '../../api_calls/api-calls';
import * as actions from '../../actions';

export class Matches extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  makeIndividualMatches = () => {
    console.log(this.props.matches);

    return this.props.matches.map((match, idx) => {
      return (
        <IndividualMatch
          key={idx}
          match={match}
        />
      );
    });
  }

  // getMatches = async () => {
  //   const matches = await fetchMatches(this.props.userId);

  //   this.props.storeMatches(matches);
  // }

  // async componentDidMount() {
  //   if (this.props.loggedIn) {
  //     this.getMatches();
  //   } else {
  //     this.props.history.push('/');
  //   }
  // }

  render() {

    return (
      <div className="matches">
        <p>These are your matches!</p>
        <div>
          {this.props.matches && this.makeIndividualMatches()}
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  userId: state.user.id,
  loggedIn: state.loggedIn,
  matches: state.matches
});

export const mapDispatchToProps = dispatch => ({
});

Matches.propTypes = {
  history: PropTypes.object.isRequired,
  matches: PropTypes.array,
  loggedIn: PropTypes.bool,
  userId: PropTypes.number,
  storeMatches: PropTypes.func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Matches));
