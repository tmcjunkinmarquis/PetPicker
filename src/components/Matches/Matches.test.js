/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import { Matches } from './Matches';

describe('Matches', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<Matches {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
