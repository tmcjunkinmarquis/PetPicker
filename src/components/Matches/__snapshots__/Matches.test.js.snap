// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Matches />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Matches
</h3>,
        <ul
          className="collection"
/>,
      ],
      "className": "matches",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Matches",
        },
        "ref": null,
        "rendered": "Matches",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "collection",
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Matches
</h3>,
          <ul
            className="collection"
/>,
        ],
        "className": "matches",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Matches",
          },
          "ref": null,
          "rendered": "Matches",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "collection",
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
